core/src/com/unciv/logic/map/tile/TileImprovementFunctions.kt

```diff
@@ -309,20 +310,28 @@ class TileImprovementFunctions(val tile: Tile) {
         val closestCity = civ.cities.minByOrNull { it.getCenterTile().aerialDistanceTo(tile) }
             ?: return
         val distance = closestCity.getCenterTile().aerialDistanceTo(tile)
-        var productionPointsToAdd = if (distance == 1) 20 else 20 - (distance - 2) * 5
-        if (tile.owningCity == null || tile.owningCity!!.civ != civ) productionPointsToAdd =
-            productionPointsToAdd * 2 / 3
-        if (productionPointsToAdd > 0) {
-            closestCity.cityConstructions.addProductionPoints(productionPointsToAdd)
+        if (distance > 5) return
+        var stats = Stats()
+        for (unique in tile.getTerrainMatchingUniques(UniqueType.ProductionBonusWhenRemoved)) {
+            stats.add(unique.stats)
+            if (unique.isModifiedByGameSpeed())
+                stats *= civ.gameInfo.speed.modifier
+        }
+        if (stats.isEmpty()) return
+        if (distance != 1) stats *= (6 - distance) / 4f
+        if (tile.owningCity == null || tile.owningCity!!.civ != civ) stats *= 2 / 3f
+        if (closestCity != null) {
```

Comment:
Closest city should *never* be null at this point

Reply:
The old code also had a check for this near the end